services-toolkit/tutorials/setup-dynamic-provisioning.hbs.md

```diff
@@ -0,0 +1,587 @@
+
+## <a id="stk-setup-dynamic-provisioning-create-rbac"></a> Create supporting RBAC
+
+There are two pieces of RBAC to consider whenever setting up a new service for dynamic provisioning in TAP. The first relates to granting permissions to the providers used in the compositions. The `Composition` created earlier uses `Object` managed resources ultimately to create `RabbitmqCluster` resources. Thus we must grant `provider-kubernetes` permission to create `RabbitmqCluster` resources. This can be done using an aggregating `ClusterRole`, as follows:
+
+Create a file named `provider-kubernetes-rmqcluster-read-writer.rbac.yml` and copy in the following contents.
+
+```yaml
+# provider-kubernetes-rmqcluster-read-writer.rbac.yml
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: rmqcluster-read-writer
+  labels:
+    services.tanzu.vmware.com/aggregate-to-provider-kubernetes: "true"
+rules:
+- apiGroups:
+  - rabbitmq.com
+  resources:
+  - rabbitmqclusters
+  verbs:
+  - "*"
+```
+
+Then use `kubectl` to apply the file to the TAP cluster.
+
+```console
+kubectl apply -f provider-kubernetes-rmqcluster-read-writer.rbac.yml
+```
+
+While not necessary here, a corresponding label `services.tanzu.vmware.com/aggregate-to-provider-helm: "true"` exists for aggregating RBAC permissions to `provider-helm` as well.
+
+The second piece of RBAC determines who is actually authorized to use the new service. This is an important piece of configuration! We are configuring an on-demand service and making it available to application teams. Without any other supporing policy in place, application teams will be able to create as many `RabbitmqClusters` as they like. This is of course the whole point of an on-demand service, however we do need to be concious of resource utilization, and we may not want just anyone who has access to the TAP cluster to be able to create new service instances on demand.
+
+Authorization can be granted using standard Kubernetes RBAC resources. Dynamic provisioning makes use of a customer RBAC verb - `claim` - which can be applied to classes in order to permit claiming from classes.
+
+Create a file named `app-operator-claim-class-bigcorp-rabbitmq.rbac.yml` and copy in the following contents.
+
+```yaml
+# app-operator-claim-class-bigcorp-rabbitmq.rbac.yml
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: app-operator-claim-class-bigcorp-rabbitmq
+  labels:
+    apps.tanzu.vmware.com/aggregate-to-app-operator-cluster-access: "true"
+rules:
+- apiGroups:
+  - services.apps.tanzu.vmware.com
+  resources:
+  - clusterinstanceclasses
+  resourceNames:
+  - bigcorp-rabbitmq
+  verbs:
+  - claim
+```
+
+Then use `kubectl` to apply the file to the TAP cluster.
+
+```console
+kubectl apply -f app-operator-claim-class-bigcorp-rabbitmq.rbac.yml
+```
+
+This `ClusterRole` grants anyone holding the `app-operator` TAP user role the ability to claim from the `bigcorp-rabbitmq` class.
```

Comment:
**suggestion** In addition to this, perhaps link off to the more thorough guide "authorize-users-to-claim...".